src/broker/content-checkpoint.c

```diff
@@ -51,45 +126,72 @@ static void checkpoint_get_continuation (flux_future_t *f, void *arg)
     flux_future_destroy (f);
 }
 
-void content_checkpoint_get_request (flux_t *h, flux_msg_handler_t *mh,
-                                     const flux_msg_t *msg, void *arg)
+static int content_checkpoint_get_backing (struct content_checkpoint *checkpoint,
+                                           const flux_msg_t *msg,
+                                           const char *key,
+                                           const char **errstr)
 {
-    struct content_checkpoint *checkpoint = arg;
     const char *topic = "content-backing.checkpoint-get";
-    const char *s = NULL;
-    const flux_msg_t *msgcpy = flux_msg_incref (msg);
     flux_future_t *f = NULL;
+    const flux_msg_t *msgref;
 
-    /* Temporarily maintain ENOSYS behavior */
-    if (checkpoint->rank == 0
-        && !content_cache_backing_loaded (checkpoint->cache)) {
-        errno = ENOSYS;
+    if (!(f = flux_rpc_pack (checkpoint->h, topic, 0, 0, "{s:s}", "key", key))
+        || flux_future_then (f,
+                             -1,
+                             checkpoint_get_continuation,
+                             checkpoint) < 0)
         goto error;
-    }
 
-    if (flux_request_decode (msg, NULL, &s) < 0)
+    msgref = flux_msg_incref (msg);
+    if (flux_future_aux_set (f,
+                             "msg",
+                             (void *)msgref,
+                             (flux_free_f)flux_msg_decref) < 0) {
+        flux_msg_decref (msgref);
```

Comment:
The `msgref` temporary variable isn't really accomplishing anything at this point.   Might as well just use 
```c
    if (flux_future_aux_set (f,
                             "msg",
                             (void *)flux_msg_incref (msg),
                             (flux_free_f)flux_msg_decref) < 0) {
        flux_msg_decref (msg);
```
Same comment applies to `put`